pkg/provider/azure_vmss_cache.go

```diff
@@ -327,6 +327,30 @@ func (ss *ScaleSet) DeleteCacheForNode(nodeName string) error {
 	return nil
 }
 
+func (ss *ScaleSet) updateCache(nodeName, resourceGroupName, vmssName, instanceID string, updatedVM *compute.VirtualMachineScaleSetVM) error {
+	cacheKey, timedCache, err := ss.getVMSSVMCache(resourceGroupName, vmssName)
+	if err != nil {
+		klog.Errorf("updateCache(%s) failed with error: %v", nodeName, err)
+		return err
+	}
+
+	localCache := &sync.Map{}
+
+	vmssVMCacheEntry := &vmssVirtualMachinesEntry{
+		resourceGroup:  resourceGroupName,
+		vmssName:       vmssName,
+		instanceID:     instanceID,
+		virtualMachine: updatedVM,
+		lastUpdate:     time.Now().UTC(),
+	}
+
+	localCache.Store(nodeName, vmssVMCacheEntry)
+
+	timedCache.Update(cacheKey, localCache)
```

Comment:
If you do this, all other VMCacheEntries in the old cache are removed. Please check function `newVMSSVirtualMachinesCache`.

Reply:
Fixed